mindsdb_native/libs/phases/model_analyzer/model_analyzer.py

```diff
@@ -120,3 +123,40 @@ def run(self):
             self.transaction.hmd['probabilistic_validators'][col] = pickle_obj(pval)
 
         self.transaction.lmd['validation_set_accuracy'] = sum(overall_accuracy_arr)/len(overall_accuracy_arr)
+
+        # conformal prediction confidence estimation
+        target = self.transaction.lmd['predict_columns'][0]
+        X = self.transaction.input_data.train_df.copy(deep=True)
+        y = X.pop(target)
+
+        data_type = self.transaction.lmd['stats_v2'][self.transaction.lmd['predict_columns'][0]]['typing']['data_type']
+        is_classification = data_type == DATA_TYPES.CATEGORICAL
+
+        fit_params = {'target': target,                               # TODO: what if n_target > 1?
+                      'columns': self.transaction.lmd['columns']}
+
+        if is_classification:
+            enc = OneHotEncoder(sparse=False)
+            enc.fit(y.values.reshape(-1, 1))
+            fit_params['one_hot_enc'] = enc
+            self.transaction.lmd['label_encoder'] = enc
+
+            adapter = ConformalClassifierAdapter
+            nc_function = MarginErrFunc()  # better than IPS as we'd need the complete distribution over all classes
+            nc_class = ClassifierNc
+            icp_class = IcpClassifier
+        else:
+            adapter = ConformalRegressorAdapter
+            nc_function = AbsErrorErrFunc()
+            nc_class = RegressorNc
+            icp_class = IcpRegressor
+
+        model = adapter(self.transaction.model_backend.predictor, fit_params=fit_params)
+        nc = nc_class(model, nc_function)
+        self.transaction.icp = icp_class(nc)
```

Comment:
This should probably be part of the `hmd` and get pickled/unpickled whenever the model is saved, to be available during the `predict` calls, unless it's never used during predict calls (It will work if the mindsdb object is never recreate it, but if you load the predictor anew it won't be saved)